### ISSUE TITLE
DefaultValue missing for @RequestBody in swagger ui 2.4

### ISSUE THREAD

**Issue by nsaxena** [springfox/springfox#1345]
After moving from com.mangofactory swagger v1.0.2 to springfox swagger v2.4, defaultValue of @ApiParam for a @RequestBody of a POST or PUT request is missing in swagger ui.

For the following method in my controller

```
@RequestMapping(method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
	@ApiOperation(value = "Create a new Project", notes = "Create a new Project.")
	@ResponseStatus(value = HttpStatus.ACCEPTED)
	@ApiResponses(value = { @ApiResponse(code = 400, message = "Bad request") })
	public Project create(@RequestBody @ApiParam(value = "Project request object", defaultValue = "{"
			+ "\n\"name\":\"Test-11\" , " + "\n\"description\":\"test Project11\" ,"
			+ "\n\"from\":{\n\"source\":\"XYZ\" , " + "\n\"sourceId\":\"123\" ," + "\n\"stack\":\"STACK\" , "
			+ "} , " + "\n\"to\": {\n\"source\":\"ABC\" , " + "\n\"sourceIdId\":\"123\" ,"
			+ "\n\"stack\":\"STACK\" , " + "} , " + "\n\"usecase\":\"one\" , "
			+ "\n\"entities\":[\"Test\"] , " + "\n\"offeringId\" :0 , "
			+ "\n\"configurationProperties\" :[{\"name\" :\"testProperty11\" , "
			+ "\n\"value\" : \"test\" , " + "\n\"id\" : 1 }," + "\n\"name\" : \"newProperty-11\" , "
			+ "\n\"value\" :\"xyz11\" , " + "\n\"id\" : null}]," + "\n \"idType\" : \"CUSTOMER\" , "
			+ "\n \"transformer\" : null , " + "\n \"status\" : null }") Project project)
			throws BaseHttpException {
```
Below swagger.json is generated which does not have a defaultValue

```
     {
            "in": "body",
            "name": "project",
            "description": "Project request object",
            "required": false,
            "schema": {
              "$ref": "#/definitions/Project"
      }
```

**Comment by dilipkrish** [comment]
Could you try in 2.5.0?

**Comment by nsaxena** [comment]
It is broken with 2.5.0 as well.

**Comment by dilipkrish** [comment]
Will take a look

**Comment by danielbcorreia** [comment]
I'm having the same issue with `@ApiImplicitParam` on a body parameter.

**Comment by dilipkrish** [comment]
`@ApiImplicitParam` is for things that are `implicit` for e.g. headers that are not on the parameter list or cookies etc.

**Comment by danielbcorreia** [comment]
Ok, right, its the same with `@ApiParam`. I actually looked into the swagger spec and when the parameter in question is a body parameter ("in": "body") the default should appear in the schema object, under the "definitions" property of the spec, instead of the "parameters" property.

So the default should be returned like this:

```javascript
"parameters": [
  {
    "in": "body",
    "name": "request",
    "description": "Request object",
    "required": true,
    "schema": {
      "$ref": "#/definitions/Request"
    }
  }
]
```

And then on the "definitions" property:

```javascript
"definitions": {
  "Request": {
    "default": "the default value",
    "example": "an example",
    "properties": { ... }
  },
}
```

See also: 
- http://swagger.io/specification/#parameterObject
- http://swagger.io/specification/#schemaObject

**Comment by marcellodesales** [comment]
* Same problem here. The body is not being loaded when providing the default value.
* Is there a way to load the JSON representation from an instance of the Input class?
 * It's too cumbersome to add an entire JSON input, specially if the input is a very large payload.

* Is there any workaround for this? I really need this ASAP :(

spring fox dependency
===

```groovy
  compile "io.springfox:springfox-swagger2:2.5.0"
  compile "io.springfox:springfox-swagger-ui:2.5.0"
```

SpringBoot Controller Config
=====

```java
@Controller
@RequestMapping("/${publisher.api.version}")
public class MatrixFileControllerSwaggerConfig {

  @Autowired
  private MatrixFileController controller;

  // stackoverflow.com/questions/352599/how-can-i-make-eclipses-autoformatter-ignore-a-section-of-code/3345289#3345289
  /* @formatter:off */
  private static final String JSON_INPUT =
      "{"
      + "\"appName\": \"publisher\","
      + "\"profilesSequence\": ["
        + "\"env\","
        + "\"dc\","
        + "\"onboard\""
       + "],"
      + "\"sigVersions\": [\"V2\"],"
      + "\"outputTypes\": [\"yml\", \"json\"],"
      + "\"profiles\": [{"
        + "\"id\": \"env\","
        + "\"descriptions\": ["
          + "\"dev\","
          + "\"qal\","
          + "\"e2e\","
          + "\"prf\","
          + "\"prd\""
         + "]"
       + "},\" {"
        + "\"id\": \"dc\","
        + "\"descriptions\": ["
          + "\"qydc\","
          + "\"lvdc\","
          + "\"aws\""
         + "]"
       + "}, {"
        + "\"id\": \"onboard\","
        + "\"descriptions\": ["
          + "\"onboard_preprod\","
          + "\"onboard_prod\""
         + "]"
       + "}]"
     + "}";
  /* @formatter:on */

  @ApiOperation(value = "Calculates the files to be generated by a given Matrix File.",
      notes = "By providing the contents of a .matrix*.json file, this api calculates all the files that will be"
          + "published in the destination.",
      consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
  @ApiResponses(value = {@ApiResponse(code = 400, message = "Bad request", reference = "When the matrix is invalid")})
  @RequestMapping(value = "/matrix/files", method = RequestMethod.POST)
  public ResponseEntity<MatrixFilesResponse> matrixFiles(
      @ApiParam(value = "Configuration Matrix", required = true,
          defaultValue = JSON_INPUT) @RequestBody ConfigMatrix configMatrix,
      @ApiParam(value = "Label / Github Branch", required = false,
          defaultValue = "master") @RequestParam(required = false) String label,
      @ApiParam(value = "S3 bucket name", required = false,
          defaultValue = "bucket-name") @RequestParam(required = false) String s3Bucket) {

    return controller.handle(configMatrix, label, s3Bucket);
  }
}
```

**Comment by marcellodesales** [comment]
@dilipkrish Any workaround for this issue?

**Comment by dilipkrish** [comment]
@marcellodesales default values are not applicable for body parameters. They are only available for form/query/cookie/header

**Comment by marcellodesales** [comment]
@dilipkrish According to http://swagger.io/specification/#parameterObject there can be one body parameter. 

* Why couldn't we provide a facility for such? 
 * Knowing the the APY takes a body, why couldn't you provide a serializer for a given instance of the body object as the default value?

**Comment by dilipkrish** [comment]
Since we use the objects provided by swagger-core you might want to create an [issue there](https://github.com/swagger-api/swagger-core).

**Comment by marcellodesales** [comment]
@dilipkrish The developer from Swagger Core commented out on this: https://github.com/swagger-api/swagger-core/issues/1844#issuecomment-246852307

> Regarding the general question - body parameters can have default values. 
> The Schema Object has a default field in it.

**Comment by dilipkrish** [comment]
@marcellodesales I responded in the ticket

**Comment by martin3361** [comment]
Why is this issue closed?
I don't think there has been a resolution to 'nsaxena's issue.
We migrated from Swagger 2.4.0 to 2.5.0 and @RequestBody annotated elements, in POST and PUT endpoints are not detected nor shown in Swagger as input parameters.
Why is that? Any configuration should be added or what?
@dilipkrish Please assist

**Comment by dilipkrish** [comment]
@martin3361 No idea whats happening, but at the very least you should try with 2.6.0 which is the latest released version. If you think there is still a problem, please submit a new issue with [more details](https://github.com/springfox/springfox/blob/master/.github/ISSUE_TEMPLATE.md)

**Comment by martin3361** [comment]
With 2.6.0 still not working, even some @PathVariable annotated parameters were invisible as well, even worse

**Comment by dilipkrish** [comment]
@martin3361 please open a new issue with **more details** so I can try to help.

**Comment by prashanth1553** [comment]
Hi
I am using spring boot. Version of springfox-swagger2 is 2.6.1 and springfox-swagger-ui is 2.6.1. Facing the same issue. Is there any solution for that?  For headers it is working fine. 

`@RequestMapping(value="/simplePost2",method= RequestMethod.POST,headers="content-type=application/json")
	@ApiOperation(value = "simplePost2", notes = "API to demonstrate post request")
	@ApiResponses({
		@ApiResponse(code = 200, message = "Success"),
		@ApiResponse(code = 404, message = "Invalid token"),
		@ApiResponse(code = 405, message = "Missing mandatory fields"),	
		@ApiResponse(code = 501, message = "Failure")})

	public org.json.simple.JSONObject simplePost2(@ApiParam(value = "json", required = true,defaultValue = "{\"id\": \"1\",\"deviceToken\":\"1\"}",name = "json")@RequestBody org.json.simple.JSONObject json,
			@ApiParam(name ="securityToken",value = "securityToken",defaultValue = "7d0fcbca-d6f0-4f29-b19f-2f8fe4e0f730" )@RequestHeader String securityToken,
			@ApiParam(name ="userId",value = "userId",defaultValue = "1" )@RequestHeader String userId) {

		//JSONObject response = new JSONObject();
		return json;
	}`

**Comment by dilipkrish** [comment]
Default values are not supported on request body parameters

**Comment by prashanth1553** [comment]
Hi @dilipkrish ,
It was working fine with com.mangofactory lib.  Then how to give some predefined value of json for request body.

**Comment by dilipkrish** [comment]
It is via examples I would presume. But default bodies are not supported. Mangofactory is based on swagger 1.2.

**Comment by prashanth1553** [comment]
In this [thread](https://github.com/springfox/springfox/issues/1536) , it mentioned like examples also won't work.

**Comment by prashanth1553** [comment]
Tried following way no luck :(

``public Driver createDriver(@RequestBody @ApiParam(examples = @Example({ @ExampleProperty("{\n" +
        "  \"ca`

**Comment by dilipkrish** [comment]
Its not implemented yet (see #853 )

**Comment by deepsaha** [comment]
Hi Everyone, I too struggled a lot setting default values for request body params initially, however Springfox provide @ApiModelProperty which can be used to set the default values. - @ApiModelProperty(required=true,value="deviceToken",example="123")

Instead of using a JSON return object if you return a POJO and annotate the POJO fields with @ApiModelProperty and set the example attribute, swagger ui and yaml will contain the default values

For example:
public org.json.simple.JSONObject simplePost2(@RequestBody POJO pojo,
			@ApiParam(name ="securityToken",value = "securityToken",defaultValue = "7d0fcbca-d6f0-4f29-b19f-2f8fe4e0f730" )@RequestHeader String securityToken,
			@ApiParam(name ="userId",value = "userId",defaultValue = "1" )@RequestHeader String userId) {

		//JSONObject response = new JSONObject();
		return json;


POJO Class:

`@ApiModel(value = "pojo", description = "Model description")
public class POJO{

	@ApiModelProperty(required=true,value="id",example="123456")
        private String id;
	
	@ApiModelProperty(required=true,value="deviceToken",example="123")
    private String deviceToken;

`

Swagger.yaml : will contain the default values under definitions

definitions:
pojo:
     type: object
    required:
      - id
      - deviceToken
    properties:
      id:
        type: string
        example: '123456'
        description: id
      deviceToken:
        type: string
        example: '123'
        description: deviceToken
    description: Model description
